### PR TITLE
[stable/redis-ha] Support explicit haproxy loadBalancerIP

### LINKED ISSUE #17047: [stable/redis-ha] haproxy loadBalancerIP field
**Is your feature request related to a problem? Please describe.**
I would like to be able to express a loadBalancerIP for the haproxy *iff* deploying haproxy as type LoadBalancer.

**Describe the solution you'd like**

Example: https://github.com/helm/charts/blob/master/stable/redis/templates/redis-with-sentinel-svc.yaml#L18

```
haproxy:
  enabled: true
  # Enable if you want a dedicated port in haproxy for redis-slaves
  readOnly:
    enabled: true
    port: 6380
  replicas: 1
  image:
    repository: haproxy
    tag: 2.0.4
    pullPolicy: IfNotPresent
  annotations: {}
  resources: {}
  ## Service type for HAProxy
  ##
  service:
    type: LoadBalancer
    loadBalancerIP: #.#.#.#
    annotations:
      service.beta.kubernetes.io/azure-load-balancer-internal: "true"
```

**Describe alternatives you've considered**
In theory, could export the yaml for the haproxy and redeploy with desired loadBalancerIP. 

**Additional context**

### PR DESCRIPTION
**Pull request by DandyDeveloper**
Signed-off-by: Aaron Layfield <aaron.layfield@gmail.com>

#### What this PR does / why we need it:
Add ability to define loadBalancerIP from service definition. 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #17047 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/redis-ha/values.yaml [hunk 1]
```diff
@@ -41,6 +41,7 @@ haproxy:
   ##
   service:
     type: ClusterIP
+    loadBalancerIP:
```
Comment (ssalaues) [inline stable/redis-ha/values.yaml:44]: I think this will have to be commented out or an empty string: `""`

Comment (DandyDeveloper) [reply stable/redis-ha/values.yaml:44]: @ssalaues I tested this, should be fine. I'll check why the test is failing...

**Pull request merged**